### ISSUE TITLE
Logs printed by the java connector node cannot be formatted correctly

### ISSUE THREAD

**Issue by StrikeW** [risingwavelabs/risingwave#14181]
### Describe the bug

Logs printed by the log4j cannot be formatted correctly after this PR https://github.com/risingwavelabs/risingwave/pull/14082.

<img width="680" alt="image" src="https://github.com/risingwavelabs/risingwave/assets/7338536/6aa944b9-0831-4774-af4b-671e63f9448f">

<img width="680" alt="image" src="https://github.com/risingwavelabs/risingwave/assets/7338536/d9c05ba9-3b26-4968-ab26-bab95d9dd167">

[grafana link](https://grafana.test.risingwave-cloud.xyz/explore?panes=%7B%22VLb%22:%7B%22datasource%22:%22PE59595AED52CF917%22,%22queries%22:%5B%7B%22refId%22:%22A%22,%22expr%22:%22%7Bapp%3D%5C%22benchmark-risingwave-compute-c-0%5C%22,namespace%3D%5C%22sysbench-cdc-20231224-200336-20231224%5C%22%7D%22,%22queryType%22:%22range%22,%22datasource%22:%7B%22type%22:%22loki%22,%22uid%22:%22PE59595AED52CF917%22%7D,%22editorMode%22:%22builder%22,%22maxLines%22:2000%7D%5D,%22range%22:%7B%22from%22:%221703448000000%22,%22to%22:%221703466659000%22%7D%7D%7D&schemaVersion=1&orgId=1)

### Expected behavior

We shoud not lost those arguments for the printed logs.

**Comment by StrikeW** [comment]
Some more findings of the problematic log 🥹

- The Java thread name is missing which is crucial for debugging
For example, the `rw-dbz-engine-runner-1001` and `debezium-postgresconnector-RW_CDC_1001-change-event-source-coordinator` in the following example before #14082. We encoded the source id into the thread name for observation.
```
2023-12-26 15:19:29,144 INFO  [rw-dbz-engine-runner-1001] common.BaseSourceTask:129 -    database.password = ********
2023-12-26 15:19:29,144 INFO  [rw-dbz-engine-runner-1001] common.BaseSourceTask:129 -    name = localhost:8432:mydb.public.RW_CDC_Sharing
2023-12-26 15:19:29,144 INFO  [rw-dbz-engine-runner-1001] common.BaseSourceTask:129 -    internal.value.converter = org.apache.kafka.connect.json.JsonConverter
2023-12-26 15:19:29,144 INFO  [rw-dbz-engine-runner-1001] common.BaseSourceTask:129 -    max.batch.size = 1024
2023-12-26 15:19:29,144 INFO  [rw-dbz-engine-runner-1001] common.BaseSourceTask:129 -    snapshot.mode = never
2023-12-26 15:19:29,144 INFO  [rw-dbz-engine-runner-1001] common.BaseSourceTask:129 -    database.history = io.debezium.relational.history.MemoryDatabaseHistory
2023-12-26 15:19:29,282 INFO  [pool-2-thread-1] jdbc.JdbcConnection:962 - Connection gracefully closed
2023-12-26 15:19:29,371 INFO  [rw-dbz-engine-runner-1001] common.BaseSourceTask:323 - No previous offsets found
2023-12-26 15:19:29,379 INFO  [rw-dbz-engine-runner-1001] postgresql.PostgresConnectorTask:109 - user 'myuser' connected to database 'mydb' on PostgreSQL 14.8 on aarch64-unknown-linux-musl, compiled by gcc (Alpine 12.2.1_git20220924-r10) 12.2.1 20220924, 64-bit with roles: ...
2023-12-26 15:19:29,386 INFO  [rw-dbz-engine-runner-1001] connection.PostgresConnection:262 - Obtained valid replication slot ReplicationSlot [active=false, latestFlushedLsn=LSN{0/EC044420}, catalogXmin=35053]
2023-12-26 15:19:29,386 INFO  [rw-dbz-engine-runner-1001] postgresql.PostgresConnectorTask:118 - No previous offset found
2023-12-26 15:19:29,386 INFO  [rw-dbz-engine-runner-1001] snapshot.NeverSnapshotter:34 - Snapshots are not allowed as per configuration, starting streaming logical changes only
2023-12-26 15:19:29,402 INFO  [rw-dbz-engine-runner-1001] util.Threads:270 - Requested thread factory for connector PostgresConnector, id = RW_CDC_1001 named = change-event-source-coordinator
2023-12-26 15:19:29,405 INFO  [rw-dbz-engine-runner-1001] util.Threads:287 - Creating thread debezium-postgresconnector-RW_CDC_1001-change-event-source-coordinator
2023-12-26 15:19:29,409 INFO  [debezium-postgresconnector-RW_CDC_1001-change-event-source-coordinator] pipeline.ChangeEventSourceCoordinator:103 - Metrics registered
2023-12-26 15:19:29,409 INFO  [debezium-postgresconnector-RW_CDC_1001-change-event-source-coordinator] pipeline.ChangeEventSourceCoordinator:106 - Context created
2023-12-26 15:19:29,412 INFO  [debezium-postgresconnector-RW_CDC_1001-change-event-source-coordinator] postgresql.PostgresSnapshotChangeEventSource:66 - According to the connector configuration no snapshot will be executed
2023-12-26 15:19:29,413 INFO  [debezium-postgresconnector-RW_CDC_1001-change-event-source-coordinator] pipeline.ChangeEventSourceCoordinator:156 - Snapshot ended with SnapshotResult [status=SKIPPED, offset=null]
2023-12-26 15:19:29,414 INFO  [debezium-postgresconnector-RW_CDC_1001-change-event-source-coordinator] pipeline.ChangeEventSourceCoordinator:236 - Connected metrics set to 'true'
```

- The log doesn't respect the logging level we defined in the [log4j configure file](https://github.com/risingwavelabs/risingwave/blob/main/java/connector-node/risingwave-connector-service/src/main/resources/log4j2.properties), which is `INFO`. The following logs are `DEBUG` log from the debezium embedded engine.

```
2023-12-26T15:06:24.843152+08:00 DEBUG risingwave_connector_node: checking for more records... class="io.debezium.connector.base.ChangeEventQueue"
2023-12-26T15:06:24.843401+08:00 DEBUG risingwave_connector_node: Embedded engine returned from polling task for records class="io.debezium.embedded.EmbeddedEngine"
2023-12-26T15:06:24.843454+08:00 DEBUG risingwave_connector_node: Received no records from the task class="io.debezium.embedded.EmbeddedEngine"
2023-12-26T15:06:24.843558+08:00 DEBUG risingwave_connector_node: Embedded engine is polling task for records on thread {} class="io.debezium.embedded.EmbeddedEngine"
2023-12-26T15:06:24.843654+08:00 DEBUG risingwave_connector_node: polling records... class="io.debezium.connector.base.ChangeEventQueue"
2023-12-26T15:06:24.843759+08:00 DEBUG risingwave_connector_node: no records available yet, sleeping a bit... class="io.debezium.connector.base.ChangeEventQueue"
2023-12-26T15:06:25.345065+08:00 DEBUG risingwave_connector_node: checking for more records... class="io.debezium.connector.base.ChangeEventQueue"
2023-12-26T15:06:25.345279+08:00 DEBUG risingwave_connector_node: Embedded engine returned from polling task for records class="io.debezium.embedded.EmbeddedEngine"
2023-12-26T15:06:25.345329+08:00 DEBUG risingwave_connector_node: Received no records from the task class="io.debezium.embedded.EmbeddedEngine"
```

**Comment by BugenZhao** [comment]
> * The Java thread name is missing which is crucial for debugging

This will be fixed with #14114.

> * The log doesn't respect the logging level we defined in the [log4j configure file](https://github.com/risingwavelabs/risingwave/blob/main/java/connector-node/risingwave-connector-service/src/main/resources/log4j2.properties?rgh-link-date=2023-12-26T07%3A27%3A10Z), which is `INFO`.

That's true. It should be now configured together with other Rust components.

https://github.com/risingwavelabs/risingwave/blob/fc6879cf22d5372059a7dcbcfe27b1dfa342a137/src/utils/runtime/src/logger.rs#L144-L186

Currently, it should follow the default level, i.e., `DEBUG` under `cfg(debug_assertions)` and `INFO` in production.
